conformance_tests/test_tiovx/test_copy_swap.c

```diff
@@ -680,8 +708,71 @@ TEST (copySwap, testCopy)
     VX_CALL(vxReleaseGraph(&graph));
 
+
     /* improve test coverage */
-    /* negative test cases for vx_pyramid */
+    /* vx_image : negative test cases */
+    vx_image image_0, image_1;
+    /* different sizes and virtual image */
+    /* width */
+    graph = vxCreateGraph(context);
+    image_0 = vxCreateImage(context, 16, 16, VX_DF_IMAGE_U8);
+    image_1 = vxCreateVirtualImage(graph, 15, 16, VX_DF_IMAGE_U8);
+    node = vxCopyNode(graph, (vx_reference)image_0, (vx_reference)image_1);
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxGetStatus((vx_reference)node));
+    EXPECT_EQ_VX_STATUS(VX_ERROR_NOT_COMPATIBLE, vxVerifyGraph(graph));
+    VX_CALL(vxReleaseImage(&image_0));
+    VX_CALL(vxReleaseImage(&image_1));
+    VX_CALL(vxReleaseNode(&node));
+    VX_CALL(vxReleaseGraph(&graph));
+    /* height */
+    graph = vxCreateGraph(context);
+    image_0 = vxCreateImage(context, 16, 16, VX_DF_IMAGE_U8);
+    image_1 = vxCreateVirtualImage(graph, 15, 16, VX_DF_IMAGE_U8);
+    node = vxCopyNode(graph, (vx_reference)image_0, (vx_reference)image_1);
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxGetStatus((vx_reference)node));
+    EXPECT_EQ_VX_STATUS(VX_ERROR_NOT_COMPATIBLE, vxVerifyGraph(graph));
+    VX_CALL(vxReleaseImage(&image_0));
+    VX_CALL(vxReleaseImage(&image_1));
+    VX_CALL(vxReleaseNode(&node));
+    VX_CALL(vxReleaseGraph(&graph));
+    /* formats */
+    graph = vxCreateGraph(context);
+    image_0 = vxCreateImage(context, 16, 16, VX_DF_IMAGE_U8);
+    image_1 = vxCreateVirtualImage(graph, 16, 16, VX_DF_IMAGE_U16);
+    node = vxCopyNode(graph, (vx_reference)image_0, (vx_reference)image_1);
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxGetStatus((vx_reference)node));
+    EXPECT_EQ_VX_STATUS(VX_ERROR_NOT_COMPATIBLE, vxVerifyGraph(graph));
+    VX_CALL(vxReleaseImage(&image_0));
+    VX_CALL(vxReleaseImage(&image_1));
+    VX_CALL(vxReleaseNode(&node));
+    VX_CALL(vxReleaseGraph(&graph));
+    graph = vxCreateGraph(context);
+    image_0 = vxCreateImage(context, 16, 16, VX_DF_IMAGE_U8);
+    image_1 = vxCreateVirtualImage(graph, 16, 16, VX_DF_IMAGE_VIRT);
+    node = vxCopyNode(graph, (vx_reference)image_0, (vx_reference)image_1);
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxGetStatus((vx_reference)node));
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxVerifyGraph(graph));
+    VX_CALL(vxReleaseImage(&image_0));
+    VX_CALL(vxReleaseImage(&image_1));
+    VX_CALL(vxReleaseNode(&node));
+    VX_CALL(vxReleaseGraph(&graph));
+
+    /* size 0 */
+    graph = vxCreateGraph(context);
+    image_0 = vxCreateImage(context, 16, 16, VX_DF_IMAGE_U8);
+    image_1 = vxCreateVirtualImage(graph, 0, 16, VX_DF_IMAGE_U8);
+    node = vxCopyNode(graph, (vx_reference)image_0, (vx_reference)image_1);
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxGetStatus((vx_reference)node));
+    EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxVerifyGraph(graph));
+    VX_CALL(vxReleaseImage(&image_1));
+    VX_CALL(vxReleaseNode(&node));
+    VX_CALL(vxReleaseGraph(&graph));
+
+    //vxReleaseImage(&array);
```

Comment:
Is this intended to be commented out?  Shall it be removed?

Reply:
yes, removed